data_model/src/asset.rs

```diff
@@ -384,6 +385,27 @@ pub struct DefinitionId {
     pub domain_id: <Domain as Identifiable>::Id,
 }
 
+/// Asset Definition Identification is represented by `name#domain_name` string.
+impl FromStr for DefinitionId {
+    type Err = ParseError;
```

Comment:
I don't like our `ParseError` it contains no useful information, because it has to contain a `'static str`. We need to come up with a different better alternative,